vignettes/rhino-fluent.Rmd

```diff
@@ -0,0 +1,556 @@
+
+```{r, echo=FALSE}
+knitr::include_graphics("images/rhino_fluent_3.png")
+```
+
+#### Render the Table from the server side inside the `datatable` module
+
+-   Till now, we did not do anything inside the server side of the `app/view/datatable.R` file.
+    Let us now use the server side to render the table rather than the UI side.
+    This is particularly helpful when we want to introduce more features inside the table - such as user inputs.
+
+-   So, here is the new modified `app/view/databale.R` file
+
+    ```         
+    box::use(
+      shiny[moduleServer, NS, uiOutput, renderUI],
+      shiny.fluent[fluentPage, Text, DetailsList],
+      tibble[tibble]
+    )
+
+    columns <- tibble(
+        fieldName = c("rep_name", "date", "deal_amount", "city", "is_closed"),
+        name = c("Sales Rep", "Date", "Deal Amount", "City", "Is Closed?"))
+
+    #' @export
+    ui <- function(id) {
+      ns <- NS(id)
+        fluentPage(
+            Text("A randomly generated dataset of imaginary sales deals"),
+            uiOutput(ns("datatable")))
+    }
+
+    #' @export
+    server <- function(id) {
+      moduleServer(id, function(input, output, session) {
+        output$datatable <- renderUI({
+            DetailsList(items = shiny.fluent::fluentSalesDeals, columns = columns)
+        })
+      })
+    }
+    ```
+
+The **`DetailsList`** table is being rendered inside the **`renderUI`** function, which generates UI output dynamically based on the **`items`** and **`columns`** arguments.
+The **`renderUI`** function is triggered whenever there is a change in the input values or reactive expressions that depend on them (we will explore this further).
+
+The **`columns`** data frame defines the field names and column names for the **`DetailsList`**.
+The **`fieldName`** column contains the names of the columns in the **`shiny.fluent::fluentSalesDeals`** data frame, and the **`name`** column contains the display names for those columns.
+
+The **`ns`** variable is used to generate namespace for the **`uiOutput`** function.
+This is important because it ensures that the input and output elements in the UI do not clash with elements from other modules or UI components in the same application.
+
+Please notice that the `uiOutput` and `renderUI` functions are from the `shiny` library, not the `shiny.fluent` library.
+Also, the `app/main.R` remains unchanged, till now.
+
+## 3. Add a simple toggle input feature for the table
+
+Till now, we could only view table with all the data it has.
+We will now add a functionality to view those rows in the table where value of the "Is Closed?" column is 1 by default and provide a toggle to the user to also include results where the value is 0.
+The "Is Closed?" column basically indicates whether the deal is closed or not with values 0 and 1 where 0 means False/No and 1 means True/Yes.
+
+-   Let us first install the `dplyr` package.
```

Comment:
Please install all required packages in a single step at the beginning of the tutorial.

Reply:
The tutorial unwinds with a theme of demand and supply, hence introduced new packages only when they were required.

Suggesting mentioning this point to install all packages at the beginning of the tutorial at the start/end of the tutorial

Reply:
If you want to add each package separately, provide complete information - remind how to install, provide updated `dependencies.R`, and finally snapshot call.